### PR TITLE
Ux improvements

### PR DESCRIPTION
**Pull request by ViAsmit**
Changes:
1. Global Theme Date for icons, button and text
2. form validation for login and signup
3. Onboarding Flow with Firestore and Firebase Storage for saving user custom info and profile image

https://user-images.githubusercontent.com/55342789/234592474-1d3d8b6b-9feb-4995-9738-66c510592df4.mp4

### CODE REVIEW

File: lib/models/resonate_user.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      uid: json["uid"],
+      userName: json["username"],
+      profileImage: json["profileImage"],
+      gender: json["gender"],
+      dateOfBirth: json["dateOfBirth"],
+      followers: List<String>.from(json["followers"] ?? []),
+      followings: List<String>.from(json["followings"]?? [])
+    );
+  }
+}
```
Comment (jddeep) [inline lib/models/resonate_user.dart:23]: Please use retrofit or freezed package.

**Review by jddeep — CHANGES_REQUESTED** [review]

File: lib/models/resonate_user.dart (hunk 1 above)
Comment (chandansgowda) [reply lib/models/resonate_user.dart:23]: Also @ViAsmit, I feel followers/following is not required for individual users. I think its better if we make it for Organizations. What do you think @jddeep ?

**Comment by chandansgowda** [comment]
Merging this PR Closes #33

File: lib/models/resonate_user.dart (hunk 1 above)
Comment (jddeep) [reply lib/models/resonate_user.dart:23]: Yes let's not dig too deep on user profiles this season @chandansgowda

**Pull request merged**